pkg/build/builder/docker.go

```diff
@@ -413,6 +419,33 @@ func traverseAST(cmd string, node *parser.Node) int {
 	return index
 }
 
+// insertEnvAfterFrom inserts an ENV Dockerfile instruction right after the
+// FROM instruction of the given dockerfile to define the environment variables
+// from env and returns the new Dockerfile.
+func insertEnvAfterFrom(env []kapi.EnvVar, dockerfile []byte) []byte {
+	if len(env) == 0 {
+		// Just return a copy.
+		return append([]byte(nil), dockerfile...)
```

Comment:
Good question. The function is supposed to return a new []byte, not the same object in memory.
Otherwise `insertEnvAfterFrom([]kapi.EnvVar{}, dockerfile)` would behave slightly different than `insertEnvAfterFrom([]kapi.EnvVar{{"FOO", "bar"}}, dockerfile)`.
If later you want to mutate the returned value, in one case you will mutate also the previous value, and in the second case you have two independent byte slices.

Look at how bytes.Join is implemented: http://docs.golang.org/src/bytes/bytes.go?s=7865:7905#L311